internal/provider/organization_resource.go

```diff
@@ -83,6 +132,58 @@ func (r *OrganizationResource) Schema(ctx context.Context, req resource.SchemaRe
 				Default:  stringdefault.StaticString(""),
 			},
 		},
+
+		Blocks: map[string]schema.Block{
+			"group_sync": schema.SingleNestedBlock{
+				Attributes: map[string]schema.Attribute{
+					"field": schema.StringAttribute{
+						Optional: true,
+						MarkdownDescription: "The claim field that specifies what groups " +
+							"a user should be in.",
+						Validators: []validator.String{
+							stringvalidator.LengthAtLeast(1),
+						},
+					},
+					"regex_filter": schema.StringAttribute{
+						Optional: true,
+						MarkdownDescription: "A regular expression that will be used to " +
+							"filter the groups returned by the OIDC provider. Any group " +
+							"not matched will be ignored.",
+						Validators: []validator.String{
+							stringvalidator.LengthAtLeast(1),
+						},
+					},
+					"auto_create_missing": schema.BoolAttribute{
```

Comment:
This defaults to `false` on coder right? Should just have a `default` set in the schema, and then you can remove the null checks on it.

Reply:
would that prevent `group_sync` from being `null` tho?

Reply:
Nope, the default only gets computed if the parent is known, so:
```
    group_sync {}
```
would default the value, but omitting the block altogether would not. If you wanted to default the block to known, and the bool to false, then you'd also add a `Default` to the block itself.